addon/components/au-dropdown.hbs

```diff
@@ -1,25 +1,36 @@
-<div class="au-c-dropdown {{this.alignment}}" ...attributes>
-  <div class="au-c-dropdown__toggle">
-    <div class="au-c-dropdown__name" data-test-dropdown-title>
+<div class="au-c-dropdown" ...attributes>
+  {{#if this.dropdownOpen}}
+  <AuButton
+    @skin={{this.skin}}
+    @size={{@size}}
+    @icon={{this.icon}}
+    @iconAlignment={{this.iconAlignment}}
+    @hideText={{@hideText}}
+    @alert={{@alert}}
+    aria-haspopup="true"
+    aria-expanded="{{if this.dropdownOpen "true" "false"}}" type="button"
+    {{on "click" this.closeDropdown}} data-test-dropdown-button>
```

Comment:
The only difference between the if and the else version seems to be the action, so I think it can be replaced with something like this?

```suggestion
    {{on "click" (if this.dropdownOpen this.closeDropdown this.openDropdown)}} data-test-dropdown-button>
```

Or alternatively with something like this if we add a toggle action:

```suggestion
    {{on "click" this.toggleDropdown}} data-test-dropdown-button>
```

Reply:
I tried to do this but creates a conflict with the focus-trap deactivate when you want to close the dropdown by clicking the button. Not sure how to solve it.

Reply:
I took a look at it and it seems we were simply sidestepping the issue by forcing a rerender with a different button component which meant the button click handler was ignored due to the teardown.

I created this [PR](https://github.com/appuniversum/ember-appuniversum/pull/324) which solves it in a different way and allows us to remove the duplication.